cpp/ql/lib/semmle/code/cpp/models/implementations/Strcat.qll

```diff
@@ -91,3 +92,69 @@ class StrcatFunction extends TaintFunction, DataFlowFunction, ArrayFunction, Sid
     buffer = true
   }
 }
+
+/**
+ * The `strlcat` function.
+ */
+class StrlcatFunction extends TaintFunction, DataFlowFunction, ArrayFunction, SideEffectFunction {
+  StrlcatFunction() {
+    this.hasGlobalName("strlcat") // strlcat(dst, src, dst_size)
+  }
+
+  /**
+   * Gets the index of the parameter that is the size of the copy (in characters).
+   */
+  int getParamSize() { exists(this.getParameter(2)) and result = 2 }
+
+  /**
+   * Gets the index of the parameter that is the source of the copy.
+   */
+  int getParamSrc() { result = 1 }
+
+  /**
+   * Gets the index of the parameter that is the destination to be appended to.
+   */
+  int getParamDest() { result = 0 }
+
+  override predicate hasDataFlow(FunctionInput input, FunctionOutput output) {
+    input.isParameter(0) and
+    output.isReturnValue()
+  }
```

Comment:
https://linux.die.net/man/3/strlcat says that the return value is "the initial length of dst plus the length of src". I don't think this should have dataflow?

Reply:
You're right. I had missed that the return value of these functions is totally different.

Reply:
Yeah, I also missed that initially. Even more reason to split this out into a separate model, I guess 🙂.

Reply:
Fixed by https://github.com/github/codeql/pull/14735/commits/5e21a5d28441db320a88cb27bf44589f44a6fb97